packages/core/src/Site/index.js

```diff
@@ -816,6 +819,59 @@ class Site {
     }
   }
 
+  async reloadSiteConfig() {
+    const oldAddressablePages = this.addressablePages.slice();
+    const oldPagesSrc = oldAddressablePages.map(page => page.src);
+    await this.readSiteConfig();
+    this.updateAddressablePages();
+    // Get pages with edited attributes but with the same src
+    const editedPages = _.differenceWith(this.addressablePages, oldAddressablePages, (newPage, oldPage) => {
+      if (!_.isEqual(newPage, oldPage)) {
+        return !oldPagesSrc.includes(newPage.src);
+      }
+      return true;
+    });
+
+    // Comparator for the _differenceWith comparison below
+    const isNewPage = (newPage, oldPage) => _.isEqual(newPage, oldPage) || newPage.src === oldPage.src;
+
+    const addedPages = _.differenceWith(this.addressablePages, oldAddressablePages, isNewPage);
+    const removedPages = _.differenceWith(oldAddressablePages, this.addressablePages, isNewPage)
+      .map(filePath => Site.setExtension(filePath.src, '.html'));
+    if (!_.isEmpty(addedPages) || !_.isEmpty(removedPages)) {
+      await this.removeAsset(removedPages);
+      await this.rebuildSourceFiles();
```

Comment:
let's use the non delayed version `this._rebuildSourceFiles();` below too for `rebuildAffectedSourceFiles`

Reply:
Alright I've updated!